tools/performance/benchmarks-data/src/index.ts

```diff
@@ -0,0 +1,210 @@
+            html_url: benchRun.html_url,
+            run_attempt: benchRun.run_attempt,
+            event: benchRun.event,
+            display_title: benchRun.display_title,
+            head_commit: {
+                id: benchRun.head_commit?.id,
+                message: benchRun.head_commit?.message,
+                timestamp: benchRun.head_commit?.timestamp,
+                author: {
+                    name: benchRun.head_commit?.author?.name,
+                }
+            }
+        },
+        label_score_dict: Object.fromEntries(results)
+    }
+    writeFile(filePath, JSON.stringify(obj, null, 2))
+    console.log("Results written to", filePath)
+}
+
+async function hasEnterpriseJob(
+    octoKit: Octokit,
+    workflowId: number
+): Promise<boolean> {
+    const allJobs = await octoKit.rest.actions.listJobsForWorkflowRun({
+        owner: 'enso-org',
+        repo: 'enso',
+        run_id: workflowId,
+        filter: "all"
+    })
+    return allJobs.data.jobs.some((job) => job.name.includes("Oracle GraalVM"))
+
+}
+
+
+async function main() {
+    program
+        .name('bench-data')
+        .description('Utility to fetch benchmarking data from GH Actions')
+        .option('-v', '--verbose', 'Enable verbose logging')
+
+    program.parse()
+
+    const options = program.opts()
+    console.log('cmdline options', options)
+
+    const ghToken = findGHToken()
+    console.log('ghToken', ghToken)
+    const octoKit = new Octokit({
+        auth: ghToken,
+    })
+    const stdLibBenchWorkflow = await octoKit.rest.actions.getWorkflow({
+        owner: 'enso-org',
+        repo: 'enso',
+        workflow_id: stdLibBenchWorkflowId,
+    })
+    console.assert(
+        stdLibBenchWorkflow.data.id === stdLibBenchWorkflowId,
+        'stdLibBenchWorkflowId mismatch'
+    )
+
+    const engineBenchWorkflow = await octoKit.rest.actions.getWorkflow({
+        owner: 'enso-org',
+        repo: 'enso',
+        workflow_id: engineBenchWorkflowId,
+    })
+    console.assert(
+        engineBenchWorkflow.data.id === engineBenchWorkflowId,
+        'engineBenchWorkflowId mismatch'
+    )
+
+    const today = new Date()
+    const beforeThreeMonths = sub(today, {months: 3})
+
+    const engineBenchRunsProm = getWorkflowRuns(octoKit, engineBenchWorkflowId, beforeThreeMonths, today)
+    const stdlibBenchRunsProm = getWorkflowRuns(octoKit, stdLibBenchWorkflowId, beforeThreeMonths, today);
+    const [engineBenchRuns, stdlibBenchRuns] = await Promise.all([engineBenchRunsProm, stdlibBenchRunsProm])
+    console.log("number of engine bench runs", engineBenchRuns.length)
+    console.log("number of stdlib bench runs", stdlibBenchRuns.length)
+
+    const engineRunsPromises = engineBenchRuns.map(async (benchRun) => {
+        if (await hasEnterpriseJob(octoKit, benchRun.id)) {
+            const results = await getEnterpriseResultsFromWorkflow(octoKit, benchRun.id);
+            writeResultsToFile(results, benchRun, "results-ee-engine");
```

Comment:
Those are async, probably want an `await` here as well (same in the next map below).